### PR TITLE
Fix main page show indefinite loading spinner when any API failed

### LINKED ISSUE #20469: Main page loading spinner spins forever on API error
I accidentally broke some of our API endpoints while working on stuff. The API endpoints return 500s but the main page just shows a loading spinner forever:

image

It seems like it should show "an error occurred" or just stop showing the loading spinner when this happens.

Errors in JS console:

image

image

To repro:

1. Change one of those endpoints to throw an Exception or otherwise trigger an artificial error when loading data for main page
2. Loads forever

### PR DESCRIPTION
**Pull request by WiNloSt**
Fixes #20469

## Repro steps
See https://github.com/metabase/metabase/issues/20469#issue-1133290892

## Before the fix
image

## After the fix
image

### CODE REVIEW

File: frontend/src/metabase/components/EmptyState.jsx [hunk 1]
```diff
@@ -29,7 +29,18 @@ const LegacyImage = props =>
       className={props.imageClassName}
     />
   ) : null;
-
+/**
```
Comment (WiNloSt) [inline frontend/src/metabase/components/EmptyState.jsx:32]: Add type to stop typescript from complaining as `EmptyState` will be used in a TypeScript file.

File: frontend/src/metabase/components/EmptyState.jsx [hunk 2]
```diff
@@ -29,7 +29,18 @@ const LegacyImage = props =>
...
+ * @typedef {Record<string, any>} EmptyStateProps
+ * @property {import("react").ReactNode} message
```
Comment (WiNloSt) [inline frontend/src/metabase/components/EmptyState.jsx:34]: Use `ReactNode` which would accept anything renderable e.g. boolean, number, string, ReactElement, etc.
```suggestion
 * @property {import("react").ReactNode} messager
```

File: frontend/src/metabase/home/homepage/components/Homepage/Homepage.tsx [hunk 3]
```diff
@@ -87,6 +92,14 @@ const Homepage = ({
             onHideSyncingModal={onHideSyncingModal}
           />
         </Fragment>
+      ) : allError ? (
+        <EmptyState
+          title={t`Something's gone wrong`}
+          message={t`We've run into an error. You can try refreshing the page, or just go back.`}
```
Comment (WiNloSt) [inline frontend/src/metabase/home/homepage/components/Homepage/Homepage.tsx:98]: Copies are copied from https://github.com/metabase/metabase/blob/42b628b0938746a880f8b6eeb90176039838cd94/frontend/src/metabase/containers/ErrorPages.jsx#L19-L20

File: frontend/src/metabase/components/EmptyState.jsx (hunk 1 above)
Comment (daltojohnso) [reply frontend/src/metabase/components/EmptyState.jsx:32]: Yep, using JSDoc here is good 👍 . I think there are a few other places where we've used it to improve type inference without moving an entire file to typescript.

**Review by daltojohnso — APPROVED** [review]

**Comment by WiNloSt** [comment]
@metabase-bot backport release-x.42.x

**Pull request merged**